routers/repo/webhook.go

```diff
@@ -861,6 +915,48 @@ func TelegramHooksEditPost(ctx *context.Context, form auth.NewTelegramHookForm)
 	ctx.Redirect(fmt.Sprintf("%s/%d", orCtx.Link, w.ID))
 }
 
+// MatrixHooksEditPost response for editing a Matrix hook
+func MatrixHooksEditPost(ctx *context.Context, form auth.NewMatrixHookForm) {
+	ctx.Data["Title"] = ctx.Tr("repo.settings")
+	ctx.Data["PageIsSettingsHooks"] = true
+	ctx.Data["PageIsSettingsHooksEdit"] = true
+
+	orCtx, w := checkWebhook(ctx)
+	if ctx.Written() {
+		return
+	}
+	ctx.Data["Webhook"] = w
+
+	if ctx.HasError() {
+		ctx.HTML(200, orCtx.NewTemplate)
+		return
+	}
+	meta, err := json.Marshal(&webhook.MatrixMeta{
+		HomeserverURL: form.HomeserverURL,
+		Room:          form.RoomID,
+		AccessToken:   form.AccessToken,
+		MessageType:   form.MessageType,
+	})
+	if err != nil {
+		ctx.ServerError("Marshal", err)
+		return
+	}
+	w.Meta = string(meta)
+	w.URL = fmt.Sprintf("%s/_matrix/client/r0/rooms/%s/send/m.room.message?access_token=%s", form.HomeserverURL, form.RoomID, form.AccessToken)
```

Comment:
The `Authorization` header is preferred to avoid the access token being logged in reverse proxies and such, but I don't know if that's easily doable with this system?

Reply:
The `Authorization` header is now used instead of the url parameter. It's not the nicest solution though.